junit-trait-runner/src/main/java/org/eclipse/collections/impl/test/junit/Junit8TraitRunnerExtension.java

```diff
@@ -14,48 +14,49 @@
 import java.lang.reflect.Method;
 import java.lang.reflect.Modifier;
 import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.HashSet;
 import java.util.LinkedList;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
 
-import org.junit.internal.MethodSorter;
-import org.junit.runners.model.FrameworkField;
-import org.junit.runners.model.FrameworkMethod;
-import org.junit.runners.model.TestClass;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
 
-public class Java8TestClass extends TestClass
+/**
+ * The class implements a custom Junit5 extension to facilitate the execution of trait-based tests
+ * that are implemented using default interface methods which Junit doesn't respect.
+ *
+ * The code manually appends the correct annotations to the test methods in the concrete classes if they are found
+ * anywhere in the interface hierarchy.
+ */
+public class Junit8TraitRunnerExtension implements BeforeEachCallback
```

Comment:
Very cool. Are you sure it's still necessary?

Years ago I started a conversation about moving this into JUnit 4 which didn't go anywhere but I was under the impression that it might be the default behavior in JUnit 5.

https://github.com/junit-team/junit4/pull/760

Reply:
Ah! You're right - you don't need it anymore so we can clean up the whole module. I will uplift soon and resubmit, thx for the catch.